### PR TITLE
website: add should-build script

### PR DESCRIPTION
**Pull request by BRKalow**
Adds `website/scripts/should-build.sh`, for use in Vercel's "Ignore Build Step" setting. More information: https://vercel.com/docs/platform/projects#ignored-build-step

[_Created by Sourcegraph campaign `bkalow/add-should-build-script`._](https://sourcegraph.hashi-mktg.com/users/bkalow/campaigns/add-should-build-script)

### CODE REVIEW

**Review by azr — APPROVED** [review]
Oh, yes, that makes sense !

File: website/scripts/should-build.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# This is run during the website build step to determine if we should skip the build or not.
+# More information: https://vercel.com/docs/platform/projects#ignored-build-step
+
+if [[ "$VERCEL_GIT_COMMIT_REF" == "stable-website"  ]] ; then
```
Comment (nywilken) [inline website/scripts/should-build.sh:6]: Does this remove the vercel deploys on master as well?

https://packer-git-master-hashicorp.vercel.app has been very useful.

Comment (BRKalow) [reply website/scripts/should-build.sh:6]: It does not! This does a diff against the previous commit to determine if anything in the website directory has changed if we are not on stable-website. This retains current behavior for main, and always deploys on stable-website.

Comment (nywilken) [reply website/scripts/should-build.sh:6]: Awesome - good stuff.

**Review by nywilken — APPROVED** [review]
LGTM

**Pull request merged**